make dist explodes

```
echo '{"version": "2.12.3", "commit": "69c74cb8dd14a27d6693f240bd686e6e23bd7623"}' > jujugui/static/gui/build/app/version.json
# We are only minifying the init bundle here because it takes considerable time.
node_modules/.bin/babel --presets babel-preset-babili --minified --no-comments ./jujugui/static/gui/build/app/init-pkg.js -o ./jujugui/static/gui/build/app/init-pkg-min.js

<--- Last few GCs --->

[30769:0x30ed7a0]   227931 ms: Mark-sweep 1282.4 (1472.1) -> 1282.4 (1473.6) MB, 1718.5 / 0.0 ms  allocation failure GC in old space requested
[30769:0x30ed7a0]   229667 ms: Mark-sweep 1282.4 (1473.6) -> 1282.3 (1433.1) MB, 1735.5 / 0.0 ms  last resort GC in old space requested
[30769:0x30ed7a0]   231443 ms: Mark-sweep 1282.3 (1433.1) -> 1282.3 (1427.6) MB, 1775.5 / 0.0 ms  last resort GC in old space requested


<--- JS stacktrace --->

==== JS stack trace =========================================

Security context: 0x2ee10b5a5879 <JSObject>
    1: set [native weak-collection.js:~27] [pc=0x3cb92cec4e29](this=0x25afa1957c19 <JSWeakMap>,key=0x1f10e18b0821 <Object map = 0x2333e2098c51>,value=0x4f5799fa6a9 <JSArray[0]>)
    2: set [/home/ubuntu/code/juju-gui/node_modules/babel-runtime/node_modules/core-js/library/modules/_collection.js:~37] [pc=0x3cb92cec567c](this=0x2f6e3858b7f9 <WeakMap map = 0x9891c3ee681>,a=0x1f10e18b0821 <Object ma...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - JavaScript heap out of memory
 1: node::Abort() [node]
 2: 0x8c20ec [node]
 3: v8::Utils::ReportOOMFailure(char const*, bool) [node]
 4: v8::internal::V8::FatalProcessOutOfMemory(char const*, bool) [node]
 5: v8::internal::Factory::NewFixedArray(int, v8::internal::PretenureFlag) [node]
 6: v8::internal::HashTable<v8::internal::ObjectHashTable, v8::internal::ObjectHashTableShape>::NewInternal(v8::internal::Isolate*, int, v8::internal::PretenureFlag) [node]
 7: v8::internal::HashTable<v8::internal::ObjectHashTable, v8::internal::ObjectHashTableShape>::New(v8::internal::Isolate*, int, v8::internal::PretenureFlag, v8::internal::MinimumCapacity) [node]
 8: v8::internal::HashTable<v8::internal::ObjectHashTable, v8::internal::ObjectHashTableShape>::EnsureCapacity(v8::internal::Handle<v8::internal::ObjectHashTable>, int, v8::internal::PretenureFlag) [node]
 9: v8::internal::ObjectHashTable::Put(v8::internal::Handle<v8::internal::ObjectHashTable>, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, int) [node]
10: v8::internal::JSWeakCollection::Set(v8::internal::Handle<v8::internal::JSWeakCollection>, v8::internal::Handle<v8::internal::Object>, v8::internal::Handle<v8::internal::Object>, int) [node]
11: v8::internal::Runtime_WeakCollectionSet(int, v8::internal::Object**, v8::internal::Isolate*) [node]
12: 0x3cb92c8842fd
Makefile:334: recipe for target 'dist' failed
make: *** [dist] Aborted (core dumped)
```

--- a/.babelrc
+++ b/.babelrc
@@ -1,7 +1,13 @@
 {
-  "presets": [],
-  "plugins": [
-    "transform-react-jsx",
-    "rewire"
-  ]
+  "env": {
+    "production": {
+      "presets": [
+        ["babili", {"mangle": false}]
+      ],
+      "plugins": [
+        "transform-react-jsx",
+        "rewire"
+      ]
+    }
+  }
 }

--- a/.babelrc
+++ b/.babelrc
@@ -1,4 +1,9 @@
 {
+  "presets": [],
+  "plugins": [
+    "transform-react-jsx",
+    "rewire"
+  ],
   "env": {
     "production": {
       "presets": [

--- a/Makefile
+++ b/Makefile
@@ -124,7 +124,7 @@ $(JUJUGUI): $(PYRAMID)
 
 $(MODULESMIN): $(NODE_MODULES) $(PYRAMID) $(BUILT_RAWJSFILES) $(MIN_JS_FILES) $(BUILT_YUI) $(BUILT_JS_ASSETS)
 	$(PY) scripts/generate_modules.py -n YUI_MODULES -s $(GUIBUILD)/app -o $(MODULES) -x "(-min.js)|(\/yui\/)"
-	$(NODE_MODULES)/.bin/babel --presets babel-preset-babili --minified --no-comments $(MODULES) -o $(MODULESMIN)
+	BABEL_ENV=production $(NODE_MODULES)/.bin/babel --minified --no-comments $(MODULES) -o $(MODULESMIN)
 
 $(GUIBUILD)/app/%.js $(GUIBUILD)/app/%-min.js: $(GUISRC)/app/%.js
 	./scripts/transpile.js
@@ -134,7 +134,7 @@ $(BUILT_JS_ASSETS): $(NODE_MODULES)
 	echo 'window.GUI_VERSION = {"version": "$(CURRENT_VERSION)", "commit": "$(CURRENT_COMMIT)"};' > $(GUIBUILD)/app/assets/javascripts/version.js
 	find $(BUILT_JS_ASSETS) -type f -name "*.js" \
 		sed s/\.js$$//g | \
-		xargs -I {} $(NODE_MODULES)/.bin/babel --presets babel-preset-babili --minified --no-comments {}.js -o {}-min.js
+		xargs -I {} BABEL_ENV=production $(NODE_MODULES)/.bin/babel --minified --no-comments {}.js -o {}-min.js
 
 $(YUI): $(NODE_MODULES)
 
@@ -332,13 +332,13 @@ version:
 .PHONY: dist
 dist: clean-all deps prod-gui test-deps collect-requirements version
 	# We are only minifying the init bundle here because it takes considerable time.
-	$(NODE_MODULES)/.bin/babel --presets babel-preset-babili --minified --no-comments ./$(GUIBUILD)/app/init-pkg.js -o ./$(GUIBUILD)/app/init-pkg-min.js
+	BABEL_ENV=production $(NODE_MODULES)/.bin/babel --minified --no-comments ./$(GUIBUILD)/app/init-pkg.js -o ./$(GUIBUILD)/app/init-pkg-min.js
 	$(PY) setup.py sdist --formats=bztar\
 
 .PHONY: fast-dist
 fast-dist: deps gui test-deps collect-requirements version
 	# We are only minifying the init bundle here because it takes considerable time.
-	$(NODE_MODULES)/.bin/babel --presets babel-preset-babili --minified --no-comments ./$(GUIBUILD)/app/init-pkg.js -o ./$(GUIBUILD)/app/init-pkg-min.js
+	BABEL_ENV=production $(NODE_MODULES)/.bin/babel --minified --no-comments ./$(GUIBUILD)/app/init-pkg.js -o ./$(GUIBUILD)/app/init-pkg-min.js
 	$(PY) setup.py sdist --formats=bztar\
 
 #######

